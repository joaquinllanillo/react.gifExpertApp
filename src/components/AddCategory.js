import React, { Fragment, useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('');

    const handdleInputChange = (e) => {
        setInputValue((e.target.value));
        console.log('handleInputChange llamado');
    }

    const handdleSubmit = (e) => {
        e.preventDefault();
        // console.log('Submit hecho');

        if(inputValue.trim().length > 2){
            setCategories(cats => [inputValue, ...cats]);
            setInputValue('');
        }
    }

    return (
        <Fragment>
            <form onSubmit = {handdleSubmit}>
                <p>{inputValue}</p>
                <input 
                    type = "text" 
                    value = {inputValue} 
                    onChange ={handdleInputChange}>
                </input>
            </form>
        </Fragment>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}