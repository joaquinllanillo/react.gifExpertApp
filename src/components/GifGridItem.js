import React from 'react'
import PropTypes from 'prop-types'

import 'animate.css';


export const GifGridItem = ({id, title, url}) => {

    console.log({id, title, url});

    return (
        <div className = "card animate__animated animate__bounce">
            <img src = {url} alt = {title} />
            <p>{title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
