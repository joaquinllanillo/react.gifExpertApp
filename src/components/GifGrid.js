import React, { Fragment } from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const { data: images, loading } = useFetchGifs(category);

    return (
        <Fragment>
            <h3 className = 'animate__animated animate__bounce'>{category}</h3>
            {loading && 'loading...'}
            {<div className='card-grid'>
                {
                    images.map(img => (
                        <GifGridItem
                            key={img.id}
                            img={img}
                            {...img}
                        />
                    ))
                }
            </div>}
        </Fragment>
    )
}
