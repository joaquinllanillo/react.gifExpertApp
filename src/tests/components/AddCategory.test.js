import React from 'react';
import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';
import '@testing-library/jest-dom';

describe('Pruebas en <AddCategory />', () => {
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories}/>);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories}/>);
    });

    test('Debe de mostrarse correctamente el componente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de cambiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value = 'Hola mundo';

        input.simulate('change', {target: {value}});
        expect(wrapper.find('p').text().trim()).toBe(value);
    });
    
    test('No debe de postear la informaciÃ³n con submit', () => {
        wrapper.find('form').simulate('submit', {preventDefault: () => {}});
        expect(setCategories).not.toHaveBeenCalled();
    });

    test('Debe de llamar el setCategories y limpiar las cajas de texto', () => {
        const value = 'Hola mundo';

        //1. Simular el Input Change
        wrapper.find('input').simulate('change', {target: {value}});

        //2. Simular el submit
        wrapper.find('form').simulate('submit', {preventDefault(){}});

        //3. setCategories se debe haber llamado
        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledTimes(1);
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));

        //4. El valor del input debe de estar ''
        expect(wrapper.find('input').prop('value')).toBe('');
    })
    
    
})
