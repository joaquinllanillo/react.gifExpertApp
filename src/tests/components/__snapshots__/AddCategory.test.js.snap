// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> Debe de mostrarse correctamente el componente 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <p />
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
